### ISSUE TITLE
Example not working?

### ISSUE THREAD

**Issue by myitcv** [adamsanderson/sexp_path#14]
``` bash
$ ruby -v
ruby 2.0.0p353 (2013-11-22 revision 43784) [x86_64-linux]
$ cat Gemfile.lock
GEM
  remote: https://rubygems.org/
  specs:
    sexp_path (0.5.2)
      sexp_processor (~> 4.2)
    sexp_processor (4.4.4)
```

With script:

``` ruby
require 'sexp_processor'
require 'sexp_path'
require 'ripper'
require 'pp'

path = ARGV.shift
code = File.read(path)
sexp = Sexp.from_array Ripper.sexp(code)

class_query = Q?{ s(:class, atom % 'class_name', _, _) }
method_query = Q?{ s(:defn, atom % 'method_name', _ ) }

results = sexp / class_query / method_query

puts path
puts "-" * 80

results.each do |sexp_result|
  class_name = sexp_result['class_name']
  method_name = sexp_result['method_name']
  puts "#{class_name}##{method_name}"
end
```

run against input file:

``` ruby
class A
  def m1
  end
end
```

Gives the following output:

``` bash
$ bundle exec ruby sexp_path.rb input.rb 
input.rb
--------------------------------------------------------------------------------
```

If you debug by looking at the `sexp` variable you'll see it is generating an S-exp, but then the `sexp_path` query always seems to return nothing.

Am I missing something obvious?

Thanks

**Comment by zenspider** [comment]
I think the example is wrong. I switched the code to ruby_parser (for my own selfish purposes) so the sexp now looks like:

```ruby
s(:class, :A, nil, s(:defn, :m1, s(:args), s(:nil))) # so much cleaner!
```

and got nothing from it. But changing the matchers to use `___` instead of `_` works:

```ruby
class_query = Q?{ s(:class, atom % 'class_name', _, ___) }
method_query = Q?{ s(:defn, atom % 'method_name', ___ ) }
```

gives:

```ruby
[{"class_name"=>:A, "method_name"=>:m1}]
```

**Comment by zenspider** [comment]
This is fixed with #16. Thanks for reporting!
